common/reflect_resource.go

```diff
@@ -41,7 +41,7 @@ var kindMap = map[reflect.Kind]string{
 // Generic interface for ResourceProvider. Using CustomizeSchema and Aliases functions to keep track of additional information
 // on top of the generated go-sdk struct. This is used to replace manually maintained structs with `tf` tags.
 type ResourceProvider interface {
-	Aliases() map[string]string
+	Aliases() map[string]map[string]string
```

Comment:
Can you add the doccomment here? What are the keys and values?

By the way, we may want to make `Aliases` optional for ResourceProvider. If a resource has no aliases, we shouldn't be forced to add an empty Aliases function, right?

Reply:
Added comments with an example.

For making `Aliases` optional, I played around with the idea of adding another interface without `Aliases()` but I think it's a bit more complicated since we also have `RecursiveResourceProvider`. I think I'm probably going to leave it as it is and try to resolve this later on. For now I think let's just merge this to unblock the next steps for the jobs migration

Reply:
Discussed offline. We'll do this.